src/tests/ftest/util/test_utils_container.py

```diff
@@ -596,6 +596,8 @@ def write_objects(self, rank=None, obj_class=None):
                 self, self.record_qty.value, self.akey_size.value,
                 self.dkey_size.value, self.data_size.value, rank, obj_class,
                 self.data_array_size.value)
+        # Mark the pool as disconnected after API operations
+        self.pool.connected = False
```

Comment:
This change requires running all tests that use `TestContainer.write_objects()` to verify it doesn't break these tests.  I'm also not sure it's the correct option.  Due to the `self.open()` call at the beginning of this method we probably should call `self.close()` here, but while the `self.open()` calls a `self.pool.connect()`, the `self.close()` does not call a `self.pool.disconnect()` - which it probably should.  There may be tests that inadvertently rely on the pool connection being left open and they would require additional changes or the `TestContainer.open()` and `TestContainer.close()` calls will need to moved outside of this method to provide that flexibility.  I may be addressing this in https://github.com/daos-stack/daos/pull/14087.